src/api/getSwap.ts

```diff
@@ -0,0 +1,103 @@
+import axios from "axios";
+import apiEndpoint from "./apiEndpoint";
+
+export interface Asset {
+  name: string;
+  quantity: string;
+}
+
+export interface Ledger {
+  name: string;
+  network: string;
+}
+
+export interface Parameters {
+  alpha_asset: Asset;
+  alpha_ledger: Ledger;
+  beta_asset: Asset;
+  beta_ledger: Ledger;
+}
+
+export enum Protocol {
+  Rfc003 = "rfc003"
+}
+
+export enum Status {
+  InProgress = "IN_PROGRESS",
+  Sent = "SWAPPED",
+  NotSwapped = "NOT_SWAPPED",
+  InternalFailure = "INTERNAL_FAILURE"
+}
+
+export enum Role {
+  Alice = "Alice",
+  Bob = "Bob"
+}
+
+export enum HtlcState {
+  NotDeployed = "NotDeployed",
+  Deployed = "Deployed",
+  Funded = "Funded",
+  Redeemed = "Redeemed",
+  Refunded = "Refunded"
+}
+
+export interface LedgerState {
+  status: HtlcState;
+  htlc_location: any;
+  deploy_tx: string;
+  fund_tx: string;
+  redeem_tx: string;
+  refund_tx: string;
+}
+
+export enum CommunicationStatus {
+  Sent = "SENT",
+  Accepted = "ACCEPTED",
+  Rejected = "REJECTED"
+}
+
+export interface CommunicationState {
+  alpha_expiry: number;
+  alpha_ledger_redeem_identity: string | null;
+  alpha_ledger_refund_identity: string;
+  beta_expiry: number;
+  beta_ledger_redeem_identity: string;
+  beta_ledger_refund_identity: string | null;
+  secret_hash: string;
+  status: CommunicationStatus;
+}
+
+export interface State {
+  alpha_ledger: LedgerState;
+  beta_ledger: LedgerState;
+  communication: CommunicationState;
+}
+
+export interface Properties {
+  parameters: Parameters;
+  protocol: Protocol;
+  status: Status;
+  role: Role;
+  state: State;
+}
+
+export interface Link {
+  rel: string[];
+  href: string;
+}
+
+export interface Swap {
+  properties: Properties;
+  links: Link[];
+}
+
+export default function getSwap(protocol: string, swapId: string) {
```

Comment:
:mega: This is exactly what you should NOT be doing :)
Please use the link provided by the comit-node to fetch the swap. Clients should not make any assumptions about what the URLs of the server look like.

Reply:
I see the problem now! Does this mean that I have to pass the URL of the swap as a query parameter to the URL of the swap page? That would look so ugly :sob:

Reply:
Good point, I am not quite sure how to solve this one.
The problem is deep-linking, e.g. when someone takes the URL from the `Swap` page and completely reloads it. The only state you get is the URL so you will need to be able to restore the page state from that.

@LLFourn had an interesting idea that might work:

You have one general, client-side route: `/show_resource` (name subject to be change). You append any server-link onto that:

- `/show_resource/swaps`
- `/show_resource/swaps/rfc003/18047108476127401`

Based on the response, the client will decide, which content to render (check the `class` property of the root-entity returned from the comit-node. it will be `swaps` for the list and `swap` for a single one).

Reply:
I think I get it. Would it be bad to use `/` as said general, client-side route. Otherwise we'd have to:
- Use a different home page to list swaps.
- Re-route (?) from `/` to `/swap_resource/swaps`.

Reply:
Redirecting seems best. At the moment the `class` property is empty for all resources though, so it's virtually impossible to tell them apart.

Also, this generic approach to fetching resources from the comit node means that the error handling should also be generic i.e. it's harder to display the correct error snackbar if you only know what you were fetching after you receive it.

Reply:
I've patched up the comit node locally and it works with your suggestion! Have a look at the changes when you can :smile:

Reply:
Oh, that one should not be empty :)
Best to open a ticket for that so we can fix it!

Good point in regards to the error handling. Would it be sufficient if we differentiate between "failed to connect" and "bad JSON" (as suggested in #59 ?). That one should be possible even if we have generic resource loading.